src/ray/common/asio/periodical_runner.cc

```diff
@@ -28,27 +28,37 @@ PeriodicalRunner::~PeriodicalRunner() {
   timers_.clear();
 }
 
-void PeriodicalRunner::RunFnPeriodically(std::function<void()> fn, uint64_t period_ms) {
+void PeriodicalRunner::RunFnPeriodically(std::function<void()> fn, uint64_t period_ms,
+                                         const std::string name) {
   if (period_ms > 0) {
     auto timer = std::make_shared<boost::asio::deadline_timer>(io_service_);
     timers_.push_back(timer);
-    DoRunFnPeriodically(fn, boost::posix_time::milliseconds(period_ms), *timer);
+    DoRunFnPeriodically(fn, boost::posix_time::milliseconds(period_ms), *timer,
+                        io_service_.get_guarded_handler_stats(name),
+                        io_service_.get_guarded_global_stats());
```

Comment:
The first invocation of `DoRunFnPeriodically` happens immediately and executes outside of the timer/event loop, so we don't specify an enqueue time.